File: web-forms/src/store/evaluation/dadosGerais.js
import { defineStore } from 'pinia'
import { ref } from 'vue';

export const dadosGeraisModel = {
    genero: null,
    complexidade: null,
    peso: null,
    altura: null,
    nome_paciente: null,
    idade: null,
    previsao_alta: null,
    dat_nascimento: null,
    cpf: null,
    rg: null,
    nome_pai: null,
    nome_mae: null,
    convenio: null,
    hospital: null,
    dat_admissao: null,
    apartamento: null,
    carteirinha: null,
    vencimento_carteirinha: null,
}

export const useDadosGeraisStore = defineStore('dadosGerais', () => {
    const dadosGerais = ref(JSON.parse(JSON.stringify(dadosGeraisModel)))
    function reset() {
        dadosGerais.value = JSON.parse(JSON.stringify(dadosGeraisModel))
    }
    return {
        dadosGerais,
        reset
    }
}, {
    persist: true
})

File: backend/internal/avaliation/domain/dados_gerais.go
package domain

type DadosGerais struct {
	Genero       string `json:"genero"` // masculino ou feminino
	Complexidade string `json:"complexidade"`

	Peso   float32 `json:"peso"` // TODO: saber se será utilizado
	Altura int     `json:"altura"`
	Idade  int     `json:"idade"`

	NomePaciente          string `json:"nome_paciente"`
	PrevisaoAlta          bool   `json:"previsao_alta"`
	DataNascimento        string `json:"data_nascimento"`
	CPF                   string `json:"cpf"`
	RG                    string `json:"rg"`
	NomePai               string `json:"nome_pai"`
	NomeMae               string `json:"nome_mae"`
	Convenio              string `json:"convenio"`
	Hospital              string `json:"hospital"`
	DataAdmissao          string `json:"dat_admissao"`
	Apartamento           string `json:"apartamento"`
	Carteirinha           string `json:"carteirinha"`
	VencimentoCarteirinha string `json:"vencimento_carteirinha"`
}
